src/status_im/contexts/shell/activity_center/notification/mentions/view.cljs

```diff
@@ -9,25 +9,45 @@
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
+(defn- mention-text
+  [literal]
+  (let [mention (rf/sub [:messages/resolve-mention literal])]
+    [quo/text
+     {:style style/mention-text
+      :size  :paragraph-1}
+     (str "@" mention)]))
+
+(defn- parsed-text->hiccup
+  [{:keys [literal destination] message-type :type}]
+  (case message-type
+    "mention" [mention-text literal]
+    "link"    destination
+    literal))
+
 (defn- message-body
-  [message]
-  (let [parsed-text          (get-in message [:content :parsed-text])
-        parsed-text-children (:children (first parsed-text))]
-    (into [quo/text
-           {:number-of-lines     1
-            :style               style/tag-text
-            :accessibility-label :activity-message-body
-            :size                :paragraph-1}]
-          (map-indexed (fn [index {:keys [type literal destination]}]
-                         ^{:key index}
-                         (case type
-                           "mention" [quo/text
-                                      {:style style/mention-text
-                                       :size  :paragraph-1}
-                                      (str "@" (rf/sub [:messages/resolve-mention literal]))]
-                           "link"    destination
-                           literal))
-                       parsed-text-children))))
```

Comment:
Refactored this component.

`map-indexed` is not needed since we are already using `into`
the metadata added is probably doing nothing since it's wrapping a `case` expression, to make it work we should use `with-meta`, but `literal` is a string, and it doesn't support `with-meta` so the code is almost doing nothing it has been simplified and split into smaller functions